kernel/frame_allocator/src/lib.rs

```diff
@@ -469,10 +472,8 @@ impl<const S: MemoryState> Drop for Frames<S> {
             MemoryState::Free => {
                 if self.size_in_frames() == 0 { return; }
         
-                let mut free_frames: FreeFrames = Frames {
-                    typ: self.typ,
-                    frames: self.frames.clone(),
-                };
+                let frames = core::mem::replace(&mut self.frames, FrameRange::empty());  
+                let free_frames: FreeFrames = Frames { typ: self.typ, frames };
         
                 let mut list = if free_frames.typ == MemoryRegionType::Reserved {
```

Comment:
this is fine for now, but in a future PR we should log an error and return early if we encounter an `Unknown` type here.